docs/ocis/adr/0028-activities-service.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "28. Activity Service"
+date: 2024-05-16T15:00:00+01:00
+weight: 28
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0028-activities-service.md
+---
+
+* Status: draft
+* Deciders: [@kobergj](https://github.com/kobergj), [@fschade](https://github.com/fschade)
+* Date: 2024-05-16
+
+## Context and Problem Statement
+
+The user should be able to see all activities for a resource.
+Besides the current resource, the user should also be able to decide if he wants to include child resource activities or not.
+
+## Decision Drivers <!-- optional -->
+
+* The user should be able to see all activities for a resource.
+* The user should be able to decide if he wants to include child resource activities.
+* Activities should be stored space efficiently.
+* Activities should be stored in a way that they can be queried efficiently.
+* Activities should stay in place event if the resource is gone.
+* Activities reflect the state at a given point in time and not the current state.
+* The Service should only store a configurable number of activities per resource.
+
+## Considered Options
+
+### Activity store
+
+* Use a go-micro store to store the individual activities.
+* Use a time series database to store the activities.
+* Use a graph database to store the activities.
+* Use a relational database to store the activities.
+* Use the file system to store the activities.
+
+### Activity format
+
+* Normalize the activities before storing them.
+* Only store relevant data to get the related event from the event-history service when needed, e.g.,
+  ```go
+    package pseudo_code
+
+    import (
+        "time"
+    )
+
+    type Activity struct {
+        ResourceId string
+        EventID string
+        Depth int64
+        Timestamp time.Time
+    }
+  ```
+* Store the activity in a human-readable way e.g. "resource A has been shared with user B."
```

Comment:
What about translations then? My naive thinking was, that an activity should have some sort of type and then we keep the message in the code (and translations in transifex).

Reply:
Exactly. That is why we didn't go for this solution. Translations need to be done live as the user could change their preferred language

Reply:
ah, good. thanks :-)